### PR TITLE
Ability to update an Opportunity

### PR DESCRIPTION
**Pull request by Himalee**
Users can select an opportunity and update one of its attributes (e.g. name, stage or cost).

### CODE REVIEW

File: src/main/java/UpdateOpportunityOptions.java [hunk 1]
```diff
@@ -1,4 +1,29 @@
...
+    private String command;
+    private String name;
+
+    UpdateOpportunityOptions(String command, String name) {
+        this.command = command;
+        this.name = name;
+    }
+
+    public static UpdateOpportunityOptions findChoice(String menuChoice) {
+        for (UpdateOpportunityOptions updateOpportunityOption : values()) {
```
Comment (ssunday) [inline src/main/java/UpdateOpportunityOptions.java]: is there anyway to use some other method to do this? Feels like this pattern is happening a lot. Or some way to refactor

Comment (Himalee) [reply src/main/java/UpdateOpportunityOptions.java]: Hi @ssunday, I have been looking into ways I can refactor this:
Enums aren't able to extend a class, so I was thinking of creating one large Enum with all possible menu choices, but adding a `menuType` to each constant indicating which type of `Menu` it belongs to e.g. starting menu.
Did you have any suggestions on how I can refactor this? 
Thanks!

Comment (ssunday) [reply src/main/java/UpdateOpportunityOptions.java]: Oh, hmmmm. Maybe a combo of enum and a map?  Like the top level key is the menuType and then the values are the options.

File: src/main/java/OptionList.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    public ArrayList<AllMenuOptions> getList() {
+        return list;
+    }
+
+    public void add(AllMenuOptions menuOption) {
+        list.add(menuOption);
+    }
+
+    public AllMenuOptions getOption(String selectedCommand) {
```
Comment (ssunday) [inline src/main/java/OptionList.java:19]: Nice!

File: src/main/java/BudgetTracker.java [hunk 3]
```diff
@@ -100,11 +100,15 @@ private void updateOpportunity() throws SQLException, ClassNotFoundException {
         if (updateOppOption == AllMenuOptions.NAME) {
             String newName = name();
             String columnName = "name";
-            databaseCommunicator.updateOpportunity(oldOpportunity, columnName, newName);
+            databaseCommunicator.updateOpportunityStringField(oldOpportunity, columnName, newName);
         } else if (updateOppOption == AllMenuOptions.DESCRIPTION) {
```
Comment (ssunday) [inline src/main/java/BudgetTracker.java]: Would a switch work better here that calls functions for each one of these?

Comment (Himalee) [reply src/main/java/BudgetTracker.java]: I have updated it and I think it works better!

File: src/main/java/PostgresCommunicator.java [hunk 4]
```diff
@@ -42,32 +37,35 @@ public Connection getConnection() throws ClassNotFoundException, SQLException {
...
     }
 
-    private void executeQuery(Connection database, String query) throws SQLException {
+    private void executeQuery(String query) throws SQLException, ClassNotFoundException {
+        Connection database = getConnection();
         Statement stmt = null;
         stmt = database.createStatement();
         stmt.executeUpdate(query);
         stmt.close();
+        database.close();
```
Comment (ssunday) [inline src/main/java/PostgresCommunicator.java:55]: Nice

**Pull request merged**